store/multi_writer_app_store.go

```diff
@@ -93,10 +116,26 @@ func NewMultiWriterAppStore(appStore *IAVLStore, evmStore *EvmStore, saveEVMStat
 		store.onlySaveEvmStateToEvmStore = bytes.Equal(store.appStore.Get(evmDBFeatureKey), []byte{1})
 	}
 
+	store.evmStateDeleted = bytes.Equal(store.appStore.Get(evmStateDeletedKey), []byte{1})
+
 	store.setLastSavedTreeToVersion(appStore.Version())
 	return store, nil
 }
 
+func (s *MultiWriterAppStore) loadOnChainConfig() {
+	configBytes := s.Get([]byte(configKey))
+	cfg := config.DefaultConfig()
+	s.numEvmKeysToPrune = int(cfg.AppStore.NumEvmKeysToPrune)
+	if len(configBytes) > 0 {
+		if err := proto.Unmarshal(configBytes, cfg); err != nil {
+			panic(err)
+		}
+	}
+	if int(cfg.AppStore.NumEvmKeysToPrune) > 0 {
```

Comment:
Why are these `int` conversions needed?

Reply:
because `RangeWithLimit` takes `int` but `NumEvmKeysToPrune` is `uint64`